producers/mysql-producer/src/main/java/com/flipkart/aesop/runtime/producer/avro/MysqlAvroEventManager.java

```diff
@@ -168,6 +194,34 @@ public int createAndAppendEvent(DbChangeEntry changeEntry, DbusEventBufferAppend
 	}
 
 	/**
+	 * This method scans new and old records and calculates changes.
+	 *
+	 * @param oldGenericRecord is a generic rocord representing old-record before change
+	 * @param newGenericRecord is a generic rocord representing new-record after change
+	 * @return HashMap<String, Object> this contains changes in form of <columnName,ColumnValue>
+	 */
+	private Map<String, Object> calculateChange(GenericRecord oldGenericRecord, GenericRecord newGenericRecord)
+	{
+		Map <String, Object> changedOldValue = new HashMap<String, Object>();
+		for (Schema.Field  field : newGenericRecord.getSchema().getFields())
+		{
+			String fieldName = field.name();
+			Object oldFieldValue = oldGenericRecord.get(fieldName);
+			Object newFieldValue = newGenericRecord.get(fieldName);
+
+			if (oldFieldValue == null && newFieldValue != null)
```

Comment:
if(oldValue != newValue && (oldValue == null || !oldValue.equals(newValue)) {
   changedOldValues.put(fieldName, oldValue);
}

Or if Guava dependency is there, use !Objects.equal(oldValue, newValue)